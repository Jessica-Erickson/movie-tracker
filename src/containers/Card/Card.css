.card {
  background-color: #CACAD1;
  border-radius: 1rem;
  box-shadow: 1rem 1rem 1rem #000000;
  display: grid;
  grid-template: 
    'a a a' auto
    'c c c' auto
    'd d d' 1fr
    'e e e' auto
    / auto auto auto;
  margin: 20px 0;
  transition-duration: .5s;
  width: 300px;
}

.card:hover {
  box-shadow: 3rem 3rem 1rem rgba(0,0,0,0.5);
  transform: scale(1.05);
  transition-duration: .5s;
}

.soda-pop {
  white-space: nowrap;
  overflow: hidden;
  text-align: center;
  text-overflow: ellipsis;
  font-size: 1.5rem;
  margin: 0;
  max-width: 200px;
}

.movie-title {
  border-radius: 0.75rem;
  display: flex;
  align-content: center;
  justify-content: space-evenly;
  grid-area: a;
  margin: 0.5rem 1rem;
  padding: 0.5rem 0;
  transition: background-color .5s, color .5s;
}

img {
  grid-area: c;
  cursor: pointer;
}

.jello {
  position: relative;
  overflow: hidden;
  height: 450px;
  width: 300px;
  z-index: 1;
}

.jello:hover::before {
  cursor: pointer;
  right: 0;
  transition: right .5s cubic-bezier(0.73, 0.11, 0.22, 0.87);
}

.jello::before {
  display: flex;
  align-items: center;
  text-align: center;
  content: 'Click to add this movie to your favorites!';
  position: absolute;
  top: 200px;
  right: 300px;
  background-color: black;
  color: #CACAD1;
  height: 50px;
  width: 300px;
  z-index: 9;
  transition: all .5s cubic-bezier(0.73, 0.11, 0.22, 0.87);
}

.overview {
  grid-area: d;
  padding: 0 1em;
}

.release_date {
  align-self: end;
  justify-self: center;
  grid-area: e;
}

.popcorn-show {
  align-self: center;
  height: 1.5rem;
  width: 1.5rem;
  transition: all .5s;
}

.popcorn-hide {
  display: none;
  transition: all .5s;
}

.has-color {
  background-color: #57282F;
  color: #CACAD1;
  transition: background-color .5s, color .5s;
}