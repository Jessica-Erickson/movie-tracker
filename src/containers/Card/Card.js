import React from 'react';
import { connect } from 'react-redux';
import { postFavorites } from '../../thunks/postFavorites';
import popcorn from '../../assets/popcorn.png';
import './Card.css'


export const Card = ({ title , id , postFavorites , poster_path , release_date , overview , user_id , vote_average, favorited }) => {
  const movie = { title , id , poster_path , release_date , overview , vote_average }
  return (
    <div className='card'>
      <div className={'movie-title ' + (favorited ? 'has-color' : '')}>
        <img
          className={'popcorn-' + (favorited ? 'show' : 'hide')} 
          src={popcorn}
          alt='Pixel popcorn. It indicates that this movie has been favorited.' />
        <h1 className='soda-pop'>{title}</h1>
        <img
          className={'popcorn-' + (favorited ? 'show' : 'hide')} 
          src={popcorn}
          alt='Pixel popcorn. It indicates that this movie has been favorited.' />
      </div>
      <div 
        className='jello'
        onClick={() => postFavorites(user_id, movie, favorited)}>
        <img 
          src={'https://image.tmdb.org/t/p/w300_and_h450_bestv2/' + poster_path} 
          alt={`This is the poster for the movie "${title}" in theaters ${release_date}.`} />
        </div>
      <p className='overview'>{overview}</p>
      <p className='release_date'>{release_date}</p>
    </div>
  )
}

export const mapStateToProps = state => {
  return { user_id: state.id }
}

export const mapDispatchToProps = dispatch => ({
  postFavorites: (id, fave, favorited) => {
    dispatch(postFavorites(id, fave, favorited))
  }
})

export default connect(mapStateToProps, mapDispatchToProps)(Card);