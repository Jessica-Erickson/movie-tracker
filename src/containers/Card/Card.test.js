import Card from './Card.js';
import React from 'react';
import { shallow } from 'enzyme';


describe('Card', () => {
  it('should match snapshot', () => {
    const movie = {
        title: 'Venom',
        poster_path: '2uNW4WbgBXL25BAbXGLnLqX71Sw.jpg',
        overview: 'This movie is RAD!',
        release_date: '2018-9-11',
        favorited: false,
      }

    const wrapper = shallow(<Card ...movie />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should call addFavorite on click', () => {

  });
});