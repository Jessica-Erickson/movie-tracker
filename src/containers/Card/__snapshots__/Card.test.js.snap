// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    movie={
      Object {
        "favorited": false,
        "overview": "This movie is RAD!",
        "poster_path": "2uNW4WbgBXL25BAbXGLnLqX71Sw.jpg",
        "release_date": "2018-9-11",
        "title": "Venom",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1 />,
        <img
          alt="This is the poster for the movie undefined in theaters undefined."
          src="https://image.tmdb.org/t/p/w300_and_h450_bestv2/undefined"
        />,
        <p />,
        <p />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "This is the poster for the movie undefined in theaters undefined.",
          "src": "https://image.tmdb.org/t/p/w300_and_h450_bestv2/undefined",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1 />,
          <img
            alt="This is the poster for the movie undefined in theaters undefined."
            src="https://image.tmdb.org/t/p/w300_and_h450_bestv2/undefined"
          />,
          <p />,
          <p />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "This is the poster for the movie undefined in theaters undefined.",
            "src": "https://image.tmdb.org/t/p/w300_and_h450_bestv2/undefined",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
